packages/devextreme/js/__internal/ui/list/m_list.base.ts

```diff
@@ -661,14 +668,27 @@ export class ListBase extends CollectionWidget<ListBaseProperties> {
     }
   }
 
-  _shouldContinueLoading(shouldLoadNextPage) {
-    const isBottomReached = getHeight(this._scrollView.content()) - getHeight(this._scrollView.container()) < (this._scrollView.scrollOffset()?.top ?? 0);
+  _shouldContinueLoading(shouldLoadNextPage: boolean): boolean {
+    const $content = this._scrollView.content();
+    const $container = this._scrollView.container();
+
+    const contentHeight = getHeight($content);
+    const containerHeight = getHeight($container);
+    const offsetTop = this._scrollView.scrollOffset()?.top;
```

Comment:
We can remove `offsetTopFinal`. I understand that many variables were introduced for better readability & maybe for debugging purpose, however I don't think we should create that much variables.
```suggestion
    const offsetTop = this._scrollView.scrollOffset()?.top ?? 0;
```

Reply:
Improved here: [f12314916353b33f10d4c7ee4c99a008bcb9d6a4]